cheetah/accelerator/element.py

```diff
@@ -91,9 +91,16 @@ def forward(self, incoming: Beam) -> Beam:
         """Forward function required by `torch.nn.Module`. Simply calls `track`."""
         return self.track(incoming)
 
-    def broadcast(self, shape: torch.Size) -> "Element":
-        """Broadcast the element to higher batch dimensions."""
-        raise NotImplementedError
+    @property
+    def batch_shape(self) -> torch.Size:
+        tensors = []
+        # Get all parameters
+        for param in self.parameters():
+            tensors.append(param)
+        # Get all buffers
+        for buffer in self.buffers():
+            tensors.append(buffer)
+        return torch.broadcast_tensors(*tensors)[0].shape
```

Comment:
Probably individual sub-classes should overwrite it if they have inherent non-scalar parameters.

Reply:
I also stumbled over this and I am reviewing it at the moment. I only saw it used in a test once. I think overwriting it would be too much effort. I'm inclined to just remove it.

Reply:
I think we can remove, and add in as necessary, it would be a good utility to have if possible